### PR TITLE
LPS-86545 doAsUserId parameter removed from URL when links to pages belonging to other sites of the portal

### PR DESCRIPTION
**Pull request by ling-alan-huang**

### CODE REVIEW

**Comment by ling-alan-huang** [comment]
Hi  @hhuijser ,

When impersonating a user, the parameter doAsUserId is added only if the url is a external url.

Thanks,
Alan

**Comment by ling-alan-huang** [comment]
ci:test

**Comment by ling-alan-huang** [comment]
Hi @hhuijser ,

The failures are not related to my changes.

One of failures is caused by _SourceCheck:JavaLongLinesCheck_ in [_PortalImpl.java_(Line:554),](https://github.com/hhuijser/liferay-portal/blob/7c41bdeafd2158212986a7443492ecd2791ac052/portal-impl/src/com/liferay/portal/util/PortalImpl.java#L548-L554) that is not my changes, and I couldn't split that single statement into two lines.

Thanks,
Alan

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java [hunk 1]
```diff
@@ -3301,7 +3302,19 @@ public String getLayoutURL(
...
-			return getLayoutActualURL(layout);
+			if (Validator.isNotNull(
+					layout.getTypeSettingsProperty(
+						LayoutTypePortletConstants.URL))) {
+
+				String url = layout.getTypeSettingsProperty(
+					LayoutTypePortletConstants.URL);
+
+				if (url.startsWith(Http.HTTP_WITH_SLASH) ||
+					url.startsWith(Http.HTTPS_WITH_SLASH)) {
```
Comment (samuelkong) [inline portal-impl/src/com/liferay/portal/util/PortalImpl.java]: Before your change, we removed `doAsUserId` from all URLs.
After your change, we only remove it from `http://` and `https://` URLs. Which means all other URL schemes are now vulnerable to LPS-79480

**Comment by ling-alan-huang** [comment]
I close this pull request, will resend.
Thanks for your catching this, @samuelkong .
cc/ @hhuijser

**Comment by ling-alan-huang** [comment]
Resend: https://github.com/hhuijser/liferay-portal/pull/3389
cc/ @samuelkong

**Pull request closed without merging**